src/components/PostComment.jsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect } from "react";
+import { useState } from "react";
+import { useParams } from "react-router"
+import { postComment } from "../api";
+
+export const CommentPoster = ({comments, setComments}) => {
+    const [name, setName] = useState([])
+    const [body, setBody] = useState([])
+    const {review_id} = useParams()
+    
+    const handlePost = (event) => {
+        event.preventDefault();
+        const newComment = {
+            author: name,
+            body:body,
+            created_at: Date(),
+            votes:0
+        }
+        setComments([...comments, newComment])
+
+        const results = [...comments]
+        postComment(review_id, name, body)
+        .catch(() => {
+            results.pop()
+            setComments(results)
+            alert("Something went wrong, do you want to try again?")
+        })
+    }
+    
+    return (
+        <section className="commentPoster">
+            <form onSubmit={handlePost}>
+            <h3>Leave your thoughts here!</h3>
+            <label htmlFor="author">
+                Username:
+            <input type="text" name="author"
+            value={name}
+            onChange={(event) => {setName(event.target.value)}}
+            key='author' 
+            placeholder="eg.jessjelly"
+            required/>
+            </label>
```

Comment:
I think a hard coded user would be better here for now rather than making me put a username